### PR TITLE
Send Slack notification upon Sidekiq job fail

### PR DESCRIPTION
**Pull request by gregorbg**
See title. Preview of how it will look in our alarms channel:
image

Keep in mind for deploy: Needs a new Vault secret

### CODE REVIEW

**Comment by danieljames-dj** [comment]
I was having plan for similar thing for WRT in future - like have a cronjob to inform WRT via Slack on number of results unposted for more than 8 hours and similar many other data.

Can we have the slack token as a column of `teams` table so that it will be easy for doing the same for WRT and in future for other teams as well?

**Comment by gregorbg** [comment]
> Can we have the slack token as a column of `teams` table so that it will be easy for doing the same for WRT and in future for other teams as well?

Hm... Right now, I am inclined to say No because the number of teams using Slack Bots will be relatively small.
If we promote Slack bots very heavily, we will suddenly face issues like Rate Limiting, API usage quotas, upload limits, etc... So that's why I don't want to push it too much.

But of course in the future we can keep this idea in mind, and your new teams/groups structure with custom metadata formats should make it relatively painless to handle this use-case :upside_down_face:

**Comment by gregorbg** [comment]
To clarify: I am not against having a WRT bot, but I think it will be sufficient to have it as a separate `AppSecrets.SLACK_WRT_BOT_TOKEN` for the time being.

**Comment by danieljames-dj** [comment]
> To clarify: I am not against having a WRT bot, but I think it will be sufficient to have it as a separate `AppSecrets.SLACK_WRT_BOT_TOKEN` for the time being.

Got it, thanks.

File: WcaOnRails/app/jobs/wca_cronjob.rb [hunk 1]
```diff
@@ -52,6 +52,13 @@ class WcaCronjob < ApplicationJob
       # Inform WST about the error so we can investigate and take action if required
       JobFailureMailer.notify_admin_of_job_failure(job, e).deliver_now
 
+      # We don't have clever mechanisms like Mailcatcher for mocking emails in non-prod
+      # environments, so for Slack we just do a hard production check instead. Shout if you have better ideas!
+      if Rails.env.production?
```
[the remark this answers is not in the document]
Comment (gregorbg) [reply WcaOnRails/app/jobs/wca_cronjob.rb]: Good point, I've added another check.

**Review by danieljames-dj — APPROVED** [review]

**Pull request merged**